engine/src/cmd/collision.cpp

```diff
@@ -156,96 +141,104 @@ void Collision::shouldApplyForceAndDealDamage(Unit* other_unit)
-    unit->ApplyForce( force-torque );
-}*/
-
-
-
-
-void Collision::dealDamage(Collision other_collision, float factor)
+// Apply damage, in VS Damage units (a unit of energy), based on change in energy from collision outcome
+void Collision::dealDamage(Collision other_collision, double deltaKE_linear, double deltaKE_angular)
 {
     if(!deal_damage)
     {
         return;
     }
 
-    // Shorten variables for better visibility
-    float& m1 = mass;
-    float& m2 = other_collision.mass;
-    Vector& v1 = velocity;
-    Vector& v2 = other_collision.velocity;
-
-    float relative_velocity = (v1-v2).Magnitude();
-    float damage = relative_velocity * relative_velocity *
-            (m2)/(m1+m2) * factor;
-
+    float damage = 0.5 * ( deltaKE_linear + deltaKE_angular) /* deltaKE is in KiloJoules, due to mass being in units of 1000 Kg, not Kg - so convert accordingly */ / configuration.physics.kilojoules_per_damage; // assign half the change in energy to this unit, convert from KJ to VSD
+    
     unit->ApplyDamage(other_collision.location.Cast(),
                       other_collision.normal,
                       damage, unit, GFXColor( 1,1,1,2 ), other_collision.unit->owner
                       != nullptr ? other_collision.unit->owner : this );
 }
 
-void Collision::applyForce(double elasticity, float& m2, Vector& v2)
+/*
+* Attempt to work around interpenetration issues[KLUDGE]
+* Specifically, impulse model for rigid body collisions assumes that, being rigid bodies, the two geometries have zero interpenetration, 
+* but VS physics granularity + object relative velocity means that non-trivial interpenetration is common
+*/
+void Collision::adjustInterpenetration(QVector& new_velocity, QVector& new_angular_velocity, const Vector& normal)
 {
-    /*
-    The current implementation is of an imperfectly elastic collision.
-    An elastic collision is an encounter between two bodies in which the total kinetic energy of the two bodies remains the same. In an ideal, perfectly elastic collision, there is no net conversion of kinetic energy into other forms such as heat, noise, or potential energy.
-    https://en.wikipedia.org/wiki/Elastic_collision
-
-    This function adds an elasticity ratio to reduce velocity, due to heat, damage, etc.
-    */
 
-    // elasticity must be between 0 and 1
-    if(elasticity >1 || elasticity <0)
+    if (!apply_force)
     {
-        BOOST_LOG_TRIVIAL(warning) << "Collision::applyForce Expected an elasticity value between 0 and 1 but got " << elasticity;
-        elasticity = std::max(elasticity, 0.0);
-        elasticity = std::min(elasticity, 1.0);
+        return;
     }
+    
+    double magnitude = new_velocity.Magnitude();
+    QVector velocity_norm = new_velocity.Normalize(); // want to perform positional displacement in direction of new velocity vector
+    double movementKludgeMagnitude = 2 * (new_angular_velocity.Dot(velocity_norm) * unit->radial_size / PI + magnitude);
```

Comment:
same with the `2`

Reply:
Agreed - albeit in this specific function everything is currently magic, so it'll just be moved to a local var just to not be nameless, rather than something more configurable.

Reply:
doesn't have to be configurable - the point of not having magic numbers is that there's no meaning in the number; using a variable (configurable) or a constant (non-configurable) value instead conveys the meaning of the value.
For now, even a simple constant right before usage is good. If we find a lot of the same, then it should be moved to a file header scope as appropriate.